libs/rtemodel/include/RteProject.h

```diff
@@ -149,6 +150,10 @@ class RteProject : public RteItem
   */
   void SetName(const std::string& name) { m_ID = name; }
 
+
+  void SetRteFolder(std::optional<std::string> rteFolder) { m_rteFolder = rteFolder; }
```

Comment:
Validation is missing!
the supplied string should : 
- be a valid folder name
- be different from folders mentioned in <files> section of the *.cprj file
Besides what should be done if project is already initialized? Copy content?

Reply:
Agreed, validation/sanitation should be added, looking at outdir is handled we should:
* Treat empty string as invalid
* Convert \ to /

Sorry, which section of folders do you refer to in the .cprj? but yes it could make sense to deny specifying rtedir == outdir || rtedir == intdir.

For me, as I update my cprj and change/set the value for the rtedir="..." I would at the same time manually also move the current RTE folder to the new name. Then the next time I run cbuildgen it would find an already initialized RTE folder (except with the new name)

So in my view cbuildgen does not need to remember the old name to try do something special in the case rtedir="..." changes.